### PR TITLE
 task: save as for model edit

### PR DESCRIPTION
**Pull request by asylves1**
# Description

- Save a new model
- Add that model to the resources panel

https://github.com/user-attachments/assets/1f5e09f2-ca7c-45d0-9a35-5d593bb216ec

Resolves #(issue)

### CODE REVIEW

File: packages/client/hmi-client/src/components/workflow/ops/model-edit/tera-model-edit.vue [hunk 1]
```diff
@@ -52,6 +52,16 @@
 				:options="outputs"
 				is-selectable
 			>
+				<section class="right-side">
+					<Button
+						class="mr-3"
+						:disabled="isSaveForReuseDisabled"
+						outlined
+						severity="secondary"
+						label="Save As Reuse"
```
Comment (shawnyama) [inline packages/client/hmi-client/src/components/workflow/ops/model-edit/tera-model-edit.vue:61]: "Save as reuse" would be the proper casing

File: packages/client/hmi-client/src/components/workflow/ops/model-edit/tera-model-edit.vue [hunk 2]
```diff
@@ -368,6 +381,17 @@ watch(
...
+	if (model) {
+		const newModel = await createModel(model);
+		if (!newModel) {
+			return;
+		}
+		amr.value = newModel;
+		createOutput(newModel);
+	}
+}
+
```
Comment (shawnyama) [inline packages/client/hmi-client/src/components/workflow/ops/model-edit/tera-model-edit.vue:394]: the `save-asset-modal` already creates a new model and adds it to your project use `save-asset.ts`, I don't think there is a need to create it here.

**Review by shawnyama — APPROVED** [review]

File: packages/client/hmi-client/src/components/workflow/ops/model-edit/tera-model-edit.vue [hunk 3]
```diff
@@ -368,6 +381,13 @@ watch(
 	{ immediate: true }
 );
 
+async function updateNodeModel(model: Model) {
```
Comment (shawnyama) [inline packages/client/hmi-client/src/components/workflow/ops/model-edit/tera-model-edit.vue:384]: no need for async

**Review by blanchco — APPROVED** [review]
okay

**Pull request merged**